### PR TITLE
Integrating rest API into user info recovery with notification in user portal

### PR DESCRIPTION
**Pull request by IndunilRathnayake**

### CODE REVIEW

File: components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/java/org/wso2/carbon/identity/mgt/endpoint/IdentityManagementEndpointConstants.java [hunk 1]
```diff
@@ -59,6 +59,15 @@
         public static final String CHALLENGE_QUESTION_2_CLAIM = "http://wso2.org/claims/challengeQuestion2";
     }
 
+    public static final class UserInfoRecovery {
+        public static final String SERVICE_CONTEXT_URL_DOMAIN = "services";
+        public static final String REST_API_URL_DOMAIN = "account-recovery";
+    }
+
+    public static final class UserInfoRecoveryStatusCodes {
+        public static final String SUCCESS = "200";
```
Comment (omindu) [inline components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/java/org/wso2/carbon/identity/mgt/endpoint/IdentityManagementEndpointConstants.java]: Can use Response.Status.OK without introducing a new var. See http://docs.oracle.com/javaee/7/api/javax/ws/rs/core/Response.Status.html#OK

File: components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/webapp/password-recovery-confirm.jsp [hunk 2]
```diff
@@ -29,6 +29,8 @@
     String fulQualifiedUsername = IdentityManagementEndpointUtil.getFullQualifiedUsername(username, tenantDomain,
                                                                                           userStoreDomain);
     if (StringUtils.isNotBlank(fulQualifiedUsername) && StringUtils.isNotBlank(confirmationKey)) {
+        request.setAttribute("username", fulQualifiedUsername);
```
Comment (omindu) [inline components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/webapp/password-recovery-confirm.jsp]: `fullQualifiedUsername` ?

File: components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/webapp/password-recovery-notify.jsp [hunk 3]
```diff
@@ -18,31 +18,42 @@
...
+
+    User user = new User();
+    user.setUserName(username);
+    user.setTenantDomain(tenantDomain);
+    user.setUserStoreDomain(userStoreDomain);
+
+    Response sendNotificationResponse = userInfoRecoveryWithNotificationClient.sendPasswordRecoveryNotification(user);
+
+    if ((sendNotificationResponse == null) || (StringUtils.isBlank(Integer.toString(sendNotificationResponse.getStatus()))) ||
+            !(Integer.toString(Response.Status.OK.getStatusCode()).equals(Integer.toString(sendNotificationResponse.getStatus())))) {
```
Comment (mefarazath) [inline components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/webapp/password-recovery-notify.jsp]: can't we simply compare the two status codes here?

File: components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/webapp/password-reset-complete.jsp [hunk 4]
```diff
@@ -16,38 +16,70 @@
...
+            resetPasswordRequest.setPassword(newPassword);
+            resetPasswordRequest.setCode(confirmationKey);
 
-        PasswordRecoverySecurityQuestionClient pwRecoverySecurityQuestionClient = new PasswordRecoverySecurityQuestionClient();
+            resetPasswordResponse = userInfoRecoveryWithNotificationClient.resetPassword(resetPasswordRequest);
 
-        ChallengeQuestionResponse challengeQuestionResponse = (ChallengeQuestionResponse)session.getAttribute("challengeQuestionResponse");
-        User user = (User) session.getAttribute("user");
+            if ((resetPasswordResponse == null) || (StringUtils.isBlank(Integer.toString(resetPasswordResponse.getStatus()))) ||
+                    !(Integer.toString(Response.Status.OK.getStatusCode()).equals(Integer.toString(resetPasswordResponse.getStatus())))) {
```
Comment (mefarazath) [inline components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/webapp/password-reset-complete.jsp]: can't we simply compare the two status codes without converting to string?

**Pull request merged**